views/default/object/plugin/details.php

```diff
@@ -0,0 +1,254 @@
+<?php
+
+$guid = (int) get_input("guid");
+
+if (empty($guid)) {
+	return;
+}
+
+$plugin = get_entity($guid);
+if (!$plugin instanceof ElggPlugin) {
+	return;
+}
+
+$active = $plugin->isActive();
+$can_activate = $plugin->canActivate();
+
+$screenshots_menu = '';
+$screenshots_body = '';
+$screenshots = $plugin->getManifest()->getScreenshots();
+if ($screenshots) {
+	
+	foreach ($screenshots as $key => $screenshot) {
+		
+		$state = "";
+		$rel = "elgg-plugin-details-screenshot-" . $key;
+		if ($key == 0) {
+			$state = " elgg-state-selected";
+		}
+		
+		$desc = elgg_echo($screenshot['description']);
+		$alt = htmlentities($desc, ENT_QUOTES, 'UTF-8');
+		
+		$thumbnail = elgg_view('output/img', array(
+			'src' => "admin_plugin_screenshot/{$plugin->getID()}/thumbnail/{$screenshot['path']}",
```

Comment:
Perhaps another PR, but we could now use simplecache to serve these thumbs rather than using a page handler